force-app/main/default/classes/DMLOperations.cls

```diff
@@ -196,11 +310,27 @@ public with sharing class DMLOperations {
      * @param leadNames The list of Lead names to be inserted.
      */
     public static void insertAndDeleteLeads(List<String> leadNames) {
-        // Loop through each Lead name
+        
+        // Initialize a list to store Lead records
+        List<Lead> leadsToInsert = new List<Lead>();
+
+            // Loop through each Lead name
+            for (String name : leadNames) {
+            // Create a new Lead
+            Lead newLead = new Lead(
+            LastName = name,
+            Company = 'Sample Company' // Adjust this based on your requirements
+            );
+
+            // Add the Lead to the list
+            leadsToInsert.add(newLead);
+        }
 
         // Insert the Leads
+        insert leadsToInsert;
 
         // Delete the Leads
+        delete leadsToInsert;
```

Comment:
good for large volumes you may find it a little more performant to skip the variable assignment in the loop and add directly to the list

```
    public static void insertAndDeleteLeads(List<String> leadNames) {
        // Loop through each Lead name
        List<Lead> leads = new List<Lead>();
        for(String leadName : leadNames){
            Lead lead = new Lead(FirstName = leadName, LastName = 'Test Lead', Company = 'Test Company');
            leads.add(lead);
        }
        insert leads;
        delete leads;
    }

```